handlers/handlers.go

```diff
@@ -424,6 +426,33 @@ func legacyLanding(w http.ResponseWriter, req *http.Request, zc ZebedeeClient, r
 		ds = append(ds, d)
 	}
 
+	// Check for filterable datasets and fetch details
+	if len(dlp.RelatedFilterableDatasets) > 0 {
+		var relatedFilterableDatasets []data.Related
+		var wg sync.WaitGroup
+		var mutex = &sync.Mutex{}
+		for _, relatedFilterableDataset := range dlp.RelatedFilterableDatasets {
+			wg.Add(1)
+			go func(ctx context.Context, dc DatasetClient, relatedFilterableDataset data.Related) {
+				defer wg.Done()
+				d, err := dc.GetByPath(ctx, relatedFilterableDataset.URI)
+				if err != nil {
+					// log error but continue to map data. any datasets that fail won't get mapped and won't be displayed on frontend
+					log.ErrorCtx(req.Context(), errors.WithMessage(err, "error fetching dataset details"), log.Data{
+						"dataset": relatedFilterableDataset.URI,
+					})
```

Comment:
trivial but might be slightly more idiomatic to have a return in this error condition so the 'happy path' is not nested in the else block